Apps/TimelineApp/core/sensor/src/main/java/uk/ac/cam/cares/jps/sensor/source/worker/BufferFlushWorker.java

```diff
@@ -0,0 +1,87 @@
+package uk.ac.cam.cares.jps.sensor.source.worker;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.hilt.work.HiltWorker;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+import org.apache.log4j.Logger;
+import org.json.JSONArray;
+
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedInject;
+import uk.ac.cam.cares.jps.sensor.source.database.SensorLocalSource;
+import uk.ac.cam.cares.jps.sensor.source.handler.SensorHandlerManager;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+
+@HiltWorker
+public class BufferFlushWorker extends Worker {
+
+    private final SensorHandlerManager sensorHandlerManager;
+    private final SensorLocalSource sensorLocalSource;
+
+    private final Logger LOGGER = Logger.getLogger(BufferFlushWorker.class);
+    private final Map<String, JSONArray> memoryBuffer = new HashMap<>();
+    private static final long THIRTY_DAYS_IN_MILLIS = 30L * 24 * 60 * 60 * 1000;
+
+    @AssistedInject
+    public BufferFlushWorker(@Assisted @NonNull Context context, @Assisted @NonNull WorkerParameters workerParams,
+                             SensorHandlerManager sensorHandlerManager,
+                             SensorLocalSource sensorLocalSource) {
+        super(context, workerParams);
+        this.sensorHandlerManager = sensorHandlerManager;
+        this.sensorLocalSource = sensorLocalSource;
+    }
+
+    @NonNull
+    @Override
+    public Result doWork() {
+        try {
+            Map<String, JSONArray> localData = sensorHandlerManager.collectSensorData();
+            for (Map.Entry<String, JSONArray> entry : localData.entrySet()) {
+                String sensorName = entry.getKey();
+                JSONArray newData = entry.getValue();
+
+                if (!memoryBuffer.containsKey(sensorName)) {
+                    memoryBuffer.put(sensorName, new JSONArray());
+                }
+
+                // Add the new data to the existing buffered data
+                JSONArray bufferedData = memoryBuffer.get(sensorName);
```

Comment:
why create local variable for `memoryBuffer.get(sensorName)`? and if want to check whether `bufferedData` is not null, it looks more proper to put outside of the for loop at line 58. 

but you have already done check and initialize the mapping in line52-54, don't think the assert is necessary.

can consider to use Map.computeIfAbsent() which get the value of key or init the key mapping if absent to make the code neat